tests/unit/store/asserts-test.js

```diff
@@ -0,0 +1,36 @@
+import {createStore} from 'dummy/tests/helpers/store';
+
+import {module, test} from 'qunit';
+
+module("unit/store/asserts - DS.Store methods produce useful assertion messages");
+
+const MODEL_NAME_METHODS = [
+  'createRecord',
+  'findRecord',
+  'findByIds',
+  'peekRecord',
+  'hasRecordForId',
+  'recordForId',
+  'query',
+  'queryRecord',
+  'findAll',
+  'peekAll',
+  'filter',
+  'recordIsLoaded',
+  'modelFor',
+  'modelFactoryFor',
```

Comment:
I looked at a few existing asserts and saw `type` so I matched that. I'll change to `modelName` and take a look for existing asserts that would be good to change as well.

I think asserts in private API can still be useful occasionally when bad values (erroneously) sneak through into internal code. If they're usually avoided in ED though I can remove them.
https://github.com/emberjs/data/pull/4178/files#r55129914